src/cascadia/TerminalApp/AppCommandlineArgs.h

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "ActionAndArgs.h"
+
+#pragma once
+#include "Commandline.h"
+
+// fwdecl unittest classes
+namespace TerminalAppLocalTests
+{
+    class CommandlineTest;
+};
+
+namespace TerminalApp
+{
+    class AppCommandlineArgs;
+};
+
+class TerminalApp::AppCommandlineArgs final
+{
+public:
+    static constexpr std::string_view NixHelpFlag{ "-?" };
+    static constexpr std::string_view WindowsHelpFlag{ "/?" };
+
+    AppCommandlineArgs();
+    ~AppCommandlineArgs() = default;
+    int ParseCommand(const Commandline& command);
+
+    static std::vector<Commandline> BuildCommands(const int argc, const wchar_t* argv[]);
+    static std::vector<Commandline> BuildCommands(winrt::array_view<const winrt::hstring>& args);
+
+    void ValidateStartupCommands();
+    std::deque<winrt::TerminalApp::ActionAndArgs>& GetStartupActions();
+    const std::string& GetExitMessage();
+
+private:
+    static const std::wregex _commandDelimiterRegex;
+
+    CLI::App _app{ "wt - the Windows Terminal" };

```

Comment:
Localizable or nah?

Reply:
As I briefly mentioned in chat before the holidays - this _can_ be localized, but it will _hyperexplode_ the unittets. In my opinion, we should probably not hyperexplode the unittests. The solution to the unittests problem appears to be "rewrite the entire thing", but I need to do a bunch more investigation of WTF is going on there. I think the right path here is to do this non-localized for now, and file a follow up for adding localization to them. That way, we won't need to include a massive "rewrite the unittests" change as a part of this already hefty PR. Thoughts?

Reply:
As long as you immediately turn around and localize the thing, I'm fine with that for the sake of making PRs easier to read. I'd be even happier if you branched off of this branch and started getting a PR ready in parallel that fixes the loc problem that we can take on the back of this one. We just definitely cannot toss localization onto a backlog item. It has to happen as close to this PR as possible.

Reply:
resolving, as we'll use the `NEEDS_LOC` to do this (and hopefully won't even need to if #4105 can get in